### PR TITLE
tkk is a new 10 key layout for a handwired diodeless keyboard

### PR DESCRIPTION
**Pull request by jfass**
tkk is a 10 key layout (alternative to ASETNIOP) on a handwired, split, diodeless (single row) keyboard using a Teensy 2.0.

## Description

New keyboard (diodeless 10-key) and layout

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by zvecr — CHANGES_REQUESTED** [review]
> tkk is a 10 key layout (alternative to ASETNIOP) on a handwired, split, diodeless (single row) keyboard using a Teensy 2.0.

You probably want to remove the duplicate copy within `keyboards/tkk`?

**Comment by jfass** [comment]
Yes! Sorry ... should be done; I believe I see it above ... or do I need to update the request?

File: keyboards/handwired/tkk/rules.mk [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# This file intentionally left blank
+
+MCU = atmega32u4
+BOOTLOADER = halfkay
+
```
Comment (zvecr) [inline keyboards/handwired/tkk/rules.mk:5]:
```suggestion
```

File: keyboards/handwired/tkk/readme.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+This is 'tkk', a ten key split keyboard wired as a single "row," ten "column" diode-less numpad. I've been dreaming of my own ten key layout since before discovering [ASETIOP](https://asetniop.com/), and [gboard's Ginny](https://www.gboards.ca/) ... but I'm too new to QMK and too rusty with C to figure out how to modify the Ginny's layout / chording engine. So here we are. The tkk [layout](https://imgur.com/a/eQzhSgw) is built around a vaguely DVORAK-ish letter layout (with some adjustments meant to aid memorization), with "modules" for numbers, movement, modifiers, and even mouse keys.
+
+* Keyboard Maintainer: [Joe Fass](https://github.com/jfass)
+* Hardware Supported: hand-wired, ten switch diode-less keyboard, using a Teensy 2.0
+* Hardware Availability: (forget where I bought the parts, but easy to find)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make tkk:default
```
Comment (zvecr) [inline keyboards/handwired/tkk/readme.md:13]:
```suggestion
    make handwired/tkk:default
```

File: keyboards/handwired/tkk/readme.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+Or:
+
+    qmk compile -kb tkk -km default
+
+In my build environment (Ubuntu), I needed to unset quite a few environment variables like CONDA\_BACKUP\_LDFLAGS that contained hardware options qmk didn't like, before this succeeded.
+
+Flashing example for this keyboard:
+
+    make tkk:default:flash
```
Comment (zvecr) [inline keyboards/handwired/tkk/readme.md:23]:
```suggestion
    make handwired/tkk:default:flash
```

File: keyboards/handwired/tkk/readme.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+* Hardware Supported: hand-wired, ten switch diode-less keyboard, using a Teensy 2.0
+* Hardware Availability: (forget where I bought the parts, but easy to find)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make tkk:default
+
+Or:
+
+    qmk compile -kb tkk -km default
```
Comment (zvecr) [inline keyboards/handwired/tkk/readme.md:17]:
```suggestion
    qmk compile -kb handwired/tkk -km default
```

File: keyboards/handwired/tkk/config.h [hunk 5]
```diff
@@ -0,0 +1,28 @@
+// Copyright 2022 Joe Fass (@jfass)
```
Comment (zvecr) [inline keyboards/handwired/tkk/config.h:1]: This file should be removed

File: keyboards/handwired/tkk/info.json [hunk 6]
```diff
@@ -0,0 +1,43 @@
+{
+    "manufacturer": "Joe Fass",
+    "keyboard_name": "tkk",
+    "maintainer": "jfass",
+    "bootloader": "atmel-dfu",
+    "comment_diode_direction": "COL2ROW",
```
Comment (zvecr) [inline keyboards/handwired/tkk/info.json:6]:
```suggestion
```

File: keyboards/handwired/tkk/info.json [hunk 7]
```diff
@@ -0,0 +1,43 @@
+{
+    "manufacturer": "Joe Fass",
+    "keyboard_name": "tkk",
+    "maintainer": "jfass",
+    "bootloader": "atmel-dfu",
```
Comment (zvecr) [inline keyboards/handwired/tkk/info.json:5]:
```suggestion
    "bootloader": "halfkay",
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by jfass** [comment]
My first GH pull request ... I _think_ I addressed all issues above?

File: keyboards/handwired/tkk/config.h (hunk 5 above)
Comment (zvecr) [reply keyboards/handwired/tkk/config.h:1]: Still needs to be removed.

File: keyboards/handwired/tkk/readme.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
+# tkk
+
+![tkk](imgur.com image replace me!)
```
Comment (zvecr) [inline keyboards/handwired/tkk/readme.md:3]: Needs to point to an image, or the line removed

File: keyboards/handwired/tkk/config.h (hunk 5 above)
Comment (jfass) [reply keyboards/handwired/tkk/config.h]: Done! .. unless I need to do something to include that last commit in the PR?

File: keyboards/handwired/tkk/readme.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
+# tkk
+
+![tkk](https://imgur.com/a/Vh43bd9)
```
Comment (drashna) [inline keyboards/handwired/tkk/readme.md:3]:
```suggestion
![tkk](https://i.imgur.com/gXAjt3Oh.jpeg)
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**